spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala

```diff
@@ -255,10 +255,20 @@ object DistriOptimizer extends AbstractOptimizer {
               val localCriterion = cached.localCriterions(i)
               val input = miniBatchBuffer(i).getInput()
               val target = miniBatchBuffer(i).getTarget()
-              val output = localModel.forward(input)
-              lossArray(i) = ev.toType[Double](localCriterion.forward(output, target))
-              val errors = localCriterion.backward(output, target)
-              localModel.backward(input, errors)
+              Engine.getEngineType() match {
+                case MklBlas =>
+                  val output = localModel.forward(input)
+                  lossArray(i) = ev.toType[Double](localCriterion.forward(output, target))
+                  val errors = localCriterion.backward(output, target)
+                  localModel.backward(input, errors)
+                case MklDnn =>
+                  Engine.computing.invokeAndWait2(Array(0).map(_ => () => {
```

Comment:
By using computing thread pool, can we make sure the underlying will use mult-omp thread ?